bin/plugin/commands/performance.js

```diff
@@ -214,24 +196,47 @@ async function runPerformanceTests( branches, options ) {
 	// 1- Preparing the tests directory.
 	log( '\n>> Preparing the tests directories' );
 	log( '    >> Cloning the repository' );
-	const baseDirectory = await git.clone( config.gitRepositoryURL );
+
+	/**
+	 * @type {string[]} git refs against which to run tests;
+	 *                  could be commit SHA, branch name, tag, etc...
+	 */
+	if ( branches.length < 2 ) {
+		throw new Error( `Need at least two git refs to run` );
+	}
+
+	const baseDirectory = getRandomTemporaryPath();
+	fs.mkdirSync( baseDirectory, { recursive: true } );
+
+	// @ts-ignore
+	const git = SimpleGit( baseDirectory );
+	await git
+		.raw( 'init' )
+		.raw( 'remote', 'add', 'origin', config.gitRepositoryURL );
+
+	for ( const branch of branches ) {
+		await git.raw( 'fetch', '--depth=1', 'origin', branch );
+	}
+
+	await git.raw( 'checkout', branches[ 0 ] );
+
 	const rootDirectory = getRandomTemporaryPath();
 	const performanceTestDirectory = rootDirectory + '/tests';
 	await runShellScript( 'mkdir -p ' + rootDirectory );
 	await runShellScript(
 		'cp -R ' + baseDirectory + ' ' + performanceTestDirectory
 	);
+
 	if ( !! options.testsBranch ) {
-		log(
-			'    >> Fetching the test branch: ' +
-				formats.success( options.testsBranch ) +
-				' branch'
-		);
-		await git.checkoutRemoteBranch(
-			performanceTestDirectory,
-			options.testsBranch
-		);
+		const branchName = formats.success( options.testsBranch );
+		log( `    >> Fetching the test-runner branch: ${ branchName }` );
+
+		// @ts-ignore
+		await SimpleGit( performanceTestDirectory )
+			.raw( 'fetch', '--depth=1', 'origin', options.testsBranch )
+			.raw( 'checkout', options.testsBranch );
```

Comment:
I guess I have the same question here :). It seems you're moving away from the "git" abstraction we had but I'm not understanding the "why"

Reply:
I had a similar question that Dennis answered [here](https://github.com/WordPress/gutenberg/pull/45255/files#r1018520922).